vendor/bat-native-ads/src/bat/ads/internal/base/strings/string_html_parse_util.cc

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/base/strings/string_html_parse_util.h"
+
+#include <regex>
+
+namespace ads {
+
+std::string FindFirstRegexMatch(const std::string& search_text,
```

Comment:
Add unit test (also for `ParseTagAttribute`)

Reply:
Unit test added in string_html_parse_util_unittest.cc